dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WebContainerRequestState.java

```diff
@@ -231,40 +235,56 @@ public void setCookieAttributes(String cookieName, String cookieAttributes) {
-        //Currently ignore all but SameSite 
+        //Currently ignore all but SameSite or Partitioned
         String[] attribute = cookieAttributes.split("=");
-        if (!attribute[0].equals("SameSite")) {
-                logger.logp(Level.FINE, CLASS_NAME, methodName, " Only SameSite attribute is supported at this time.");
+        if (!(attribute[0].equals("SameSite") || attribute[0].equals("Partitioned"))) {
+                logger.logp(Level.FINE, CLASS_NAME, methodName, " Only SameSite attribute and Partitioned attribute are supported at this time.");
                 return;
         }
+
+        // Should we throw any exceptions? such as: 
+        // if(attribute.length == 1) { // No attributeValue set
+        //     throw IllegalStateException("attribute value not specififed");
+        // }
         
         if (cookieAttributesMap == null) {
-            cookieAttributesMap = new HashMap<String,String>();
+            cookieAttributesMap = new HashMap<String,HashMap<String,String>>();
+            cookieAttributesMap.put(cookieName, new HashMap<String,String>() {{
+                put(attribute[0], attribute[1]);
+            }});
+        } else {
+            HashMap<String,String> existingAttributesMap = cookieAttributesMap.get(cookieName);
+            existingAttributesMap.put(attribute[0], attribute[1]);
+            cookieAttributesMap.put(cookieName, existingAttributesMap);
         }
-
-        cookieAttributesMap.put(cookieName, cookieAttributes);
     }
     
     /**
      * Return the Cookie attributes associated with the provided cookieName that were
      * added via the setCookieAttributes()
+     * 
+     * If multiple attributes exist for a cookie, then they will be returned semi-colon delimited key value pairs.
+     * For example: attributeName1=attributeValue1;attributeName2=attributeValue2
+     * 
+     * If no cookie attributes exist, null is returned;
      *
      * @param cookieName - The name of the Cookie the attributes were set for.
      * @return - The Cookie attributes associated with the specified Cookie name.
      */
     public String getCookieAttributes(String cookieName) {
         String methodName = "getCookieAttributes";
 
-        if (cookieAttributesMap == null) {
+        if (cookieAttributesMap == null || cookieAttributesMap.get(cookieName) == null) {
                 return null;
         }
 
         if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable (Level.FINE)) {
             logger.logp(Level.FINE, CLASS_NAME, methodName, " cookieName --> " + cookieName);
         }
 
-        String cookieAttributes = cookieAttributesMap.get(cookieName);
+        String cookieAttributes = cookieAttributesMap.get(cookieName).entrySet()
+                                                     .stream().map(e->e.getKey() + "=" + e.getValue())
+                                                     .collect(java.util.stream.Collectors.joining(";"));
```

Comment:
We're using Java 8 as a min, so this I opted to use stream as it's a lot easier make the return value.